### PR TITLE
Add ErrorReporter unexpected method

### PR DESCRIPTION
**Pull request by bdewater**
### Type of Change

- [ ] Add RBI for a new gem
- [x] Modify RBI for an existing gem
- [ ] Other: 

### Changes

* Gem name: Active Support
* Gem version: 7.2 beta
* Gem source: https://github.com/rails/rails/blob/9098f535b2accef27c6c284dede119c936c44595/activesupport/lib/active_support/error_reporter.rb#L122-L154

### CODE REVIEW

File: rbi/annotations/activesupport.rbi
Comment (egiurleo) [inline rbi/annotations/activesupport.rbi:476]: I think you win the award for the very first version annotation? 🥇

Comment (KaanOzkan) [reply rbi/annotations/activesupport.rbi:476]: How do we want to handle runtime checks? I think CI will use the latest version however beta versions can only be specified in the Gemfile?

There's also the case of annotations with an incompatible version, `< 7.2.0beta1`. Should we just use `@method_missing` instead of dealing with this?

**Comment by bdewater** [comment]
With Rails 7.2 out this now passes CI.

File: rbi/annotations/activesupport.rbi
Comment (KaanOzkan) [reply rbi/annotations/activesupport.rbi:497]: I think this PR is fine thanks to https://github.com/Shopify/tapioca/pull/1585 but I'll leave it to @egiurleo before merging. We may wanna test versioning with apps that use lower Rails versions as we merge this.

Comment (egiurleo) [reply rbi/annotations/activesupport.rbi:497]: Yeah let's merge this and I'll add a note to our meeting on Monday to talk about it.

**Review by egiurleo — APPROVED** [review]

**Pull request merged**
